tinytodo/src/context.rs

```diff
@@ -355,27 +363,126 @@ impl AppContext {
 
     #[tracing::instrument(skip(policy_set))]
     fn update_policy_set(&mut self, policy_set: PolicySet) -> Result<AppResponse> {
-        let policies = rename_from_id_annotation(policy_set)?;
-        self.policies = policies;
-        info!("Reloaded policy set");
+        let mut new_policies = rename_from_id_annotation(policy_set)?;
+        let mut err = false;
+        // for each existing template-linked policy,
+        //   link against the new version of the template in the new policy set if present
+        for p in self.policies.policies() {
+            match p.template_id() {
+                None => (), // not a template-linked policy
+                Some(tid) => {
+                    // template-linked policy
+                    match new_policies.template(tid) {
+                        None => {
+                            // template not in new policy set
+                            let tidx = tid.clone();
+                            let pidx = p.id().clone();
+                            err = true;
+                            error!("Error when reloading policies: Could not find policy template {tidx} to link {pidx}")
+                        }
+                        Some(_) => {
+                            // found template in new policy set
+                            match p.template_links() {
+                                None => error!("Error when reloading policies: Template with no matching links"),
```

Comment:
Should this set `err = true;`?

Reply:
This is an internal error, which should never happen. I.e., you should not have `template_id()` return `Some(...)` but then have `template_links()` return `None`. I'll change this to use `expect` and produce a `panic!` if it ever happens.

Reply:
Thought: We could change the `Policy::template_links()` API to return both the linked values and the template ID, together in an `Option`. Then we wouldn't need multiple calls. You could just call `template_links()` as a way to test whether a policy is template-linked, and if it is to return `Some(tid,links)`. That would simplify a bunch of logic here, and avoid the need for app code to assume the Cedar-internal invariant.

It would be changing the API, but we haven't released the `PolicySet::template_links()` API yet, so it won't be a breaking change.

Reply:
https://github.com/cedar-policy/cedar/issues/548 -- I'm of a mixed mind about whether doing this is worth it.